py-polars/polars/expr/whenthen.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import polars.functions as F
+from polars.expr.expr import Expr
+from polars.utils._parse_expr_input import parse_as_expression
+from polars.utils._wrap import wrap_expr
+
+if TYPE_CHECKING:
+    from polars.polars import PyExpr
+    from polars.type_aliases import IntoExpr
+
+
+class When:
+    """Utility class. See the `when` function."""
+
+    def __init__(self, pywhen: Any):
+        self._pywhen = pywhen
+
+    def then(self, expr: IntoExpr) -> WhenThen:
+        """
+        Values to return in case of the predicate being `True`.
+
+        See Also
+        --------
+        pl.when : Documentation for `when, then, otherwise`
+
+        """
+        expr = parse_as_expression(expr, str_as_lit=True)
+        pywhenthen = self._pywhen.then(expr)
+        return WhenThen(pywhenthen)
+
+
+class WhenThen(Expr):
+    """Utility class. See the `when` function."""
+
+    def __init__(self, pywhenthen: Any):
+        self._pywhenthen = pywhenthen
+
+    @classmethod
+    def _from_pyexpr(cls, pyexpr: PyExpr) -> Expr:  # type: ignore[override]
+        return wrap_expr(pyexpr)
+
+    @property
+    def _pyexpr(self) -> PyExpr:
+        return self._pywhenthen.otherwise(F.lit(None)._pyexpr)
```

Comment:
By inheriting Expr, overriding the classmethod constructor, and adding the _pyexpr as a property, we can make it behave like an Expr with the additional methods when/otherwise.